modules/ting_field_search/ting_field_search.module

```diff
@@ -432,39 +433,202 @@ function ting_field_search_ting_search_settings_submit($form, &$form_state) {
-  if (!variable_get('ting_field_search_search_style', FALSE)) {
-    return;
+function ting_field_search_preprocess_html(&$vars) {
+  $profiles = ting_field_search_profiles_load_all();
+
+  foreach ($profiles as $profile) {
+    // Expended search box not checked go to the next profile.
+    if (!array_key_exists('search_box', $profile->config['user_interaction']) || !$profile->config['user_interaction']['search_box']) {
+      continue;
+    }
+
+    // If the profile has visibility all always expand the search field.
+    if ($profile->config['user_interaction']['visibility'] === 'all') {
+      $vars['classes_array'][] = 'extended-search-is-open';
+      break;
+    }
+
+    $visibility = $profile->config['user_interaction']['visibility'] == 'expect' ? 0 : 1;
+    $pages = $profile->config['user_interaction']['pages'];
+    $path = drupal_strtolower(drupal_get_path_alias($_GET['q']));
+
+    // We have an optimization hack to speed up search (by-pass panels 2x
+    // times loading). So all searches are submitted to "/search-blank",
+    // which need to be added to pages to ensure no form validation errors.
+    // If visibility is "expect" it should not be added to the pages list.
+    if ($visibility) {
+      $pages .= "\r\nsearch-blank";
+    }
+
+    // Compare the lowercase internal and lowercase path alias (if any).
+    $page_match = drupal_match_path($path, $pages);
+    if ($path != $_GET['q']) {
+      $page_match = $page_match || drupal_match_path($_GET['q'], $pages);
+    }
+
+    // When $visibility has a value of 0, the profile is displayed on
+    // all pages except those listed in $pages. When set to 1, it
+    // is displayed only on those pages listed in $pages.
+    if (!($visibility xor $page_match)) {
+      $vars['classes_array'][] = 'extended-search-is-open';
+      break;
+    }
   }
+}
 
-  // Dont't alter the form if this is the node search page.
-  if (strpos(current_path(), 'search/node') === 0) {
-    return;
+
+/**
+ * Find the default profile for the current context.
+ *
+ * @return string
+ *   The name of the profile that is default.
+ */
+function ting_field_search_get_default_profile() {
+  // If there's an active profile and it is stick; set it as default value.
+  if ($default_profile = ting_field_search_get_active_profile()) {
+    $default_profile = $default_profile->config['user_interaction']['sticky'] ? $default_profile->name : '';
+  }
+
+  if (empty($default_profile)) {
+    $profiles = ting_field_search_profiles_load_all();
+    foreach ($profiles as $profile) {
+      // Expended search box not checked go to the next profile.
+      if (!array_key_exists('default_profile', $profile->config['user_interaction']) || !$profile->config['user_interaction']['default_profile']) {
+        continue;
+      }
+
+      // If the profile has visibility all always expand the search field.
+      if ($profile->config['user_interaction']['visibility'] === 'all') {
```

Comment:
Do we need to check for the existence of the `visibility` entry here?

Reply:
I don't remeber why, but a lot of testing (by GBA) has gone into this and there was a lot og issue with frontpage vs. other pages etc....